src/main/java/com/cburch/logisim/std/io/LedBarHDLGeneratorFactory.java

```diff
@@ -28,18 +28,47 @@
 
 package com.cburch.logisim.std.io;
 
+import java.util.ArrayList;
+
 import com.cburch.logisim.data.Attribute;
+import com.cburch.logisim.data.AttributeSet;
 import com.cburch.logisim.data.BitWidth;
+import com.cburch.logisim.fpga.designrulecheck.Netlist;
+import com.cburch.logisim.fpga.designrulecheck.NetlistComponent;
+import com.cburch.logisim.fpga.hdlgenerator.AbstractHDLGeneratorFactory;
+import com.cburch.logisim.fpga.hdlgenerator.HDL;
+import com.cburch.logisim.fpga.hdlgenerator.HDLGeneratorFactory;
+
+public class LedBarHDLGeneratorFactory extends AbstractHDLGeneratorFactory {
 
-public class LedBarHDLGeneratorFactory extends DotMatrixHDLGeneratorFactory {
+  protected Attribute<BitWidth> getAttributeColumns() {
+    return LedBar.ATTR_MATRIX_COLS;
+  }
 
   @Override
-  protected Attribute<BitWidth> getAttributeRows() {
-    return LedBar.ATTR_MATRIX_ROWS;
+  public ArrayList<String> GetInlinedCode(Netlist netlist, Long componentId, NetlistComponent componentInfo, String circuitName) {
```

Comment:
Newly added metho names must start lowercased (to distinguish from class/interface name) -> `getInlinedCode()`. We still got tons of legacy names but let's work on reducing the number of these if possible (so I do not complain on overriden methods for that reason).

Reply:
You're right, this should be done in a refactoring pass (which will be huge)

Reply:
Here, however it overrides an existing one, and I cannot change it